spec/models/presenters/pipeline_info_presenter_spec.rb

```diff
@@ -575,6 +575,24 @@ def stub_labware_find_all_barcode(labwares)
         expect(presenter.grandparent_purposes?).to be true
       end
     end
+
+    context 'when an error occurs during API calls' do
+      let(:parent_purpose) { create(:purpose, name: 'Parent Purpose') }
+
+      let(:parent_tube) { create(:stock_tube, purpose: parent_purpose, uuid: 'parent-tube-uuid') }
+
+      before do
+        allow(labware).to receive_messages(parents: [parent_tube])
+        stub_labware_find_all_barcode([parent_tube])
+
+        allow(Sequencescape::Api::V2::Labware).to receive(:find_all)
+          .and_raise(JsonApiClient::Errors::ClientError.new(500, 'Internal Server Error'))
+      end
+
+      it 'returns true' do
+        expect(presenter.grandparent_purposes?).to be true
+      end
+    end
```

Comment:
OK and this just controls whether there's a `...` in app/views/labware/_pipeline_info.erb, so not a big deal if it's incorrect!